examples/blake2f.rs

```diff
@@ -0,0 +1,1459 @@
+use chiquito::{
+    frontend::dsl::{
+        cb::{eq, select, table},
+        lb::LookupTable,
+        super_circuit, CircuitContext, StepTypeSetupContext, StepTypeWGHandler,
+    },
+    plonkish::{
+        backend::halo2::{chiquitoSuperCircuit2Halo2, ChiquitoHalo2SuperCircuit},
+        compiler::{
+            cell_manager::{MaxWidthCellManager, SingleRowCellManager},
+            config,
+            step_selector::SimpleStepSelectorBuilder,
+        },
+        ir::sc::SuperCircuit,
+    },
+    poly::ToExpr,
+    sbpir::query::Queriable,
+};
+use halo2_proofs::{
+    dev::MockProver,
+    halo2curves::{bn256::Fr, group::ff::PrimeField},
+};
+use std::hash::Hash;
+
+pub const IV_LENGTH: u64 = 8;
+pub const SIGMA_VECTOR_LENGTH: u64 = 16;
+pub const SIGMA_VECTOR_NUMBER: u64 = 10;
+pub const R1: u64 = 32;
+pub const R2: u64 = 24;
+pub const R3: u64 = 16;
+pub const R4: u64 = 63;
+pub const MIXING_ROUND: u64 = 12;
+pub const BITS_NUMBER: u64 = 16;
```

Comment:
@10to4 any number os a number. Perhaps if this is the bit width of something, BITS_WIDTH is more clear.

Reply:
It is something else it seems. Like the amount of values possible for 4bits. I am not sure what name would be more clear for this

Reply:
I need to think about the best name for this contant, and also if it makes sense to separate it from VALUE_4BITS

Reply:
I want to use this constant to represent the number of 4-bit values that are separated from a original value.